pkg/server/combined_statement_stats.go

```diff
@@ -722,55 +717,50 @@ SELECT
 FROM (SELECT fingerprint_id,
              app_name,
              max(aggregated_ts)                                     AS aggregated_ts,
-             %s AS metadata,
+             `, metadataAggFn, ` AS metadata,
              merge_statement_stats(statistics)                      AS statistics
       FROM %s %s
       GROUP BY
           fingerprint_id,
-          app_name) %s %s`, metadataAggFn, CrdbInternalStmtStatsCached, aostClause, whereClause, orderAndLimit), args...)
+          app_name) %s
+%s`}, "")
 
-		if err != nil {
-			return nil, srverrors.ServerError(ctx, err)
-		}
-	}
+	var it isql.Rows
+	var err error
+	defer func() {
+		err = closeIterator(it, err)
+	}()
 
-	// If there are no results from the activity table, retrieve the data from the persisted table.
-	if it == nil || !it.HasResults() {
-		if it != nil {
-			err = closeIterator(it, err)
-		}
+	switch r.stmtSourceTable {
+	case CrdbInternalStmtStatsCached:
 		it, err = getIterator(
 			ctx,
-			ie,
-			queryFormat,
-			CrdbInternalStmtStatsPersisted,
-			"combined-stmts-persisted-by-interval",
+			r.ie,
+			// The statement activity table has aggregated metadata.
+			activityQuery,
+			CrdbInternalStmtStatsCached,
+			"combined-stmts-activity-by-interval",
```

Comment:
It looks like `getIterator` already prefies the opName with `console-combined-stmts`, so maybe we can delete that part here. I'm not a big fan of using this getIterator thin wrapper since all it really does is format the query and pass 
 everything off to the internal executor, but I guess we should limit the amount of refactoring here.